lib/types/duration.js

```diff
@@ -161,6 +156,7 @@ Duration.fromBuffer = function (buffer) {
  * @returns {Duration}
  */
 Duration.fromString = function (input) {
+  // This does a lot of stuff. Is it nesesery to implement it in rust, or should we just keep it as is?
   const isNegative = input.charAt(0) === '-';
   const source = isNegative ? input.substr(1) : input;
   if (source.charAt(0) === 'P') {
```

Comment:
Does the question still stand, or have we decided that we're keeping this logic in JS?

Reply:
The question still stands